scripts/Engineering/gui/engineering_diffraction/tabs/calibration/model.py

```diff
@@ -0,0 +1,243 @@
+                                      OutputParametersTableName=table_name)
+        return output
+
+    @staticmethod
+    def calculate_vanadium_correction(vanadium_path):
+        """
+        Runs the vanadium correction algorithm.
+        :param vanadium_path: The path to the vanadium data.
+        :return: The integrated workspace and the curves generated by the algorithm.
+        """
+        try:
+            Load(Filename=vanadium_path, OutputWorkspace=VANADIUM_INPUT_WORKSPACE_NAME)
+        except Exception as e:
+            logger.error("Error when loading vanadium sample data. "
+                         "Could not run Load algorithm with vanadium run number: " +
+                         str(vanadium_path) + ". Error description: " + str(e))
+            raise RuntimeError
+        EnggVanadiumCorrections(VanadiumWorkspace=VANADIUM_INPUT_WORKSPACE_NAME,
+                                OutIntegrationWorkspace=INTEGRATED_WORKSPACE_NAME,
+                                OutCurvesWorkspace=CURVES_WORKSPACE_NAME)
+        Ads.remove(VANADIUM_INPUT_WORKSPACE_NAME)
+        integrated_workspace = Ads.Instance().retrieve(INTEGRATED_WORKSPACE_NAME)
+        curves_workspace = Ads.Instance().retrieve(CURVES_WORKSPACE_NAME)
+        return integrated_workspace, curves_workspace
+
+    def create_output_files(self, calibration_dir, difc, tzero, ceria_path, vanadium_path,
+                            instrument):
+        """
+        Create output files from the algorithms in the specified directory
+        :param calibration_dir: The directory to save the files into.
+        :param difc: DIFC values from the calibration algorithm.
+        :param tzero: TZERO values from the calibration algorithm.
+        :param ceria_path: The path to the ceria data file.
+        :param vanadium_path: The path to the vanadium data file.
+        :param instrument: The instrument (ENGINX or IMAT)
+        """
+        if not path.exists(calibration_dir):
+            makedirs(calibration_dir)
+        filename, vanadium_no, ceria_no = self._generate_output_file_name(vanadium_path,
+                                                                          ceria_path,
+                                                                          instrument,
+                                                                          bank="all")
+        # Both Banks
+        file_path = calibration_dir + filename
+        write_ENGINX_GSAS_iparam_file(file_path,
+                                      difc,
+                                      tzero,
+                                      ceria_run=ceria_no,
+                                      vanadium_run=vanadium_no)
+        # North Bank
+        file_path = calibration_dir + self._generate_output_file_name(
+            vanadium_path, ceria_path, instrument, bank="north")[0]
+        write_ENGINX_GSAS_iparam_file(file_path, [difc[0]], [tzero[0]],
```

Comment:
At some point we will need to generalise this for IMAT support. We'll most likely want to put this in instrument parameters files but for now we can leave this hard coded. Could you move it out to a module-scope variable so it's more obvious from the top of the file that this is hard coded?